### PR TITLE
[ASV-2214] Rakam user properties

### PR DESCRIPTION
**Pull request by phillipsgoncalves**
**What does this PR do?**

   Sending all rakam user properties at once as rakam is paid by events, so if we sent all properties in different "events" we would be spending more.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] FirstLaunchAnalytics.java

**How should this be manually tested?**

  Run the app and check in Charles that the events are only sent once. 

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-2214](https://aptoide.atlassian.net/browse/ASV-2214)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java [hunk 1]
```diff
@@ -334,12 +327,23 @@ private void setUTMDimensionsToUnknown() {
     FlurryAgent.addSessionProperty(UTM_CAMPAIGN, UNKNOWN);
     FlurryAgent.addSessionProperty(UTM_CONTENT, UNKNOWN);
     FlurryAgent.addSessionProperty(ENTRY_POINT, UNKNOWN);
+  }
+
+  private void sendRakamUserProperties(String utmContent, String utmSource, String utmCampaign,
+      String utmMedium, String entryPoint, String packageName) {
+    Identify rakamProperties = new Identify();
+
+    rakamProperties.set(GMS_RAKAM, rakamGmsValueCache);
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java]: rakamGmsValueCache might be null. It depends in another method being called, which is called in AnalyticsActivity... this should be reviewed.

File: app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java [hunk 2]
```diff
@@ -218,8 +219,12 @@ private void setupRakamFirstLaunchSuperProperty(boolean isFirstLaunch) {
   private void setupDimensions(android.app.Application application) {
     if (!checkForUTMFileInMetaINF(application)) {
       setUTMDimensionsToUnknown();
+      sendRakamUserProperties(UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN, UNKNOWN,
+          application.getPackageName());
     } else {
       setUserProperties(utmSource, utmMedium, utmCampaign, utmContent, entryPoint);
+      sendRakamUserProperties(utmContent, utmSource, utmCampaign, utmMedium, entryPoint,
+          application.getPackageName());
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java]: why don't we set them in the setUserProperties anyway? shouldn't the setUserProperties deal with all the user properties from different analytics?

**Comment by jdandrade** [comment]
We are only reducing like 3 calls to 1. Shouldn't we check if we can send a batch of events with a single call? like we do for legacy BI?

File: app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java (hunk 1 above)
Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java]: Created a wrapper that will get the value from the adnetworks (regarding gms). Started using it the first time i got it and will calculate it if it does not exist. I am using that "singleton" on both places.

File: app/src/main/java/cm/aptoide/pt/GmsStatusValueProvider.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package cm.aptoide.pt;
+
+import android.content.Context;
+import cm.aptoide.pt.dataprovider.ads.AdNetworkUtils;
+
+public class GmsStatusValueProvider {
+
+  private static final String HAS_HGMS = "Has GMS";
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/GmsStatusValueProvider.java]: typo here?

File: app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java (hunk 2 above)
Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java]: I didnt want to merge both methods to keep rakam events sending independent

Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java]: They can be independent, but UserProperties is a parent of RakamUserProperties, all names should be at the same level

**Pull request merged**